### ISSUE TITLE
C++ code feature: CMakeLists.txt template

### ISSUE THREAD

**Issue by WetzelVictor** [pyphs/pyphs#99]
Hi pyphs users!

I've been having some troubles with th C++ simulation feature.

### Informations
* `pyphs` version: master branch on GitHub
* `cmake` version: 3.5.1
* `Ubuntu` version: 16.04.4 LTS Xenial Xerus

Everything is explained in the attached notebook.

### Issue
Me and my colleagues had problems with the C++ code feature.
After some research I've determined that the CMakeLists.txt was responsible.

When the C++ 11 feature is activated:
 ```

    target_compile_features(<project_name> PUBLIC cxx_std_11)

 ```

CMake throws an error: this feature is unknown:
```

    -- The CXX compiler identification is GNU 5.4.0
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    CMake Error at CMakeLists.txt:31 (target_compile_features):
      target_compile_features specified unknown feature "cxx_std_11" for target
      "l".
    ```

    -- Configuring incomplete, errors occurred!
    See also "/home/victorw/git/vocal-phs/python/output/l/CMakeFiles/CMakeOutput.log".

```

### Fix
I've changed the `CMakeLists.txt` template. Here is the link to the commit, on my own fork of pyphs:

    https://github.com/WetzelVictor/pyphs/commit/badbc7cf396f9d4cf8f991bc07dea4c9e3f34f1e


I've replaced the `target_compile_features(<project_name> PUBLIC cxx_std_11)` by `set (CMAKE_CXX_STANDARD 11)`



What are your insights on this matter? Did I forget to mention some informations?

Thank you all 
[cpp_code_genration_example.tar.gz](https://github.com/pyphs/pyphs/files/1766337/cpp_code_genration_example.tar.gz)

**Comment by afalaize** [comment]
Thank you.
This is a retrograde change compared to [this commit](https://github.com/pyphs/pyphs/commit/8506e056b90f1dfed9fb915e2a5a680f2840e3a6#diff-bfb2cc9ee357c715fdeca9a6c4aa233b). I have not a clear idea on the rational underlying this change. I will ask the author to clarify.

**Comment by FabricioS** [comment]
Thanks Antoine for the quick answer.
FYI, Eigen is C++98 compliant. Moreover, it is a header only library, it is theoretically not concerned by the way the standard is set.

**Comment by afalaize** [comment]
Yes. The need for cxx1 spec is related to the use of "dynamic initialization" of eigen matrices in the c++ code generated by PyPHS (see [here](https://github.com/pyphs/pyphs/blob/d6fa231cfb9043c9507371041956618ae50ed6ac/pyphs/numerics/cpp/functions.py#L279)). This is somehow related to [this stack overflow answer](https://stackoverflow.com/a/37913974). This is a choice of implementation which, as everything in PyPHS, is not a definitive choice. Any proposal on this point is welcome.
I will consider the merge of PR #100 this week-end.

**Comment by WetzelVictor** [comment]
Someone also answered on my StackOverflow [question](https://stackoverflow.com/questions/49012222/c11-activation-with-target-compile-feature-or-set-cmake-cxx-standard/49027994#49027994)

> The cxx_std_11 compiler meta-feature is not available in your version of CMake. it was introduced in  3.8, so this would explain the error, CMake 3.8 release notes.

So either the `CMakeLists.txt` should request a higher `Cmake` version (3.8 which is not delivered directly in stock Ubuntu 16.04 LTS distros for instance), or we retrograde to the mentionned commit in order to have backward compatibility with older versions of `Cmake`..

**Comment by FabricioS** [comment]
Thanks Antoine for considering the pull request.
There is no trouble per se with the use of C++11 standard, just with the way to specify it to cmake (and to the compiler) or to ensure cmake has the requested feature.

The remaining question is whether there was a rationale for Remy to suggest that modification in his commit

**Comment by WetzelVictor** [comment]
Issue solved with PR #101 and #104

**Comment by afalaize** [comment]
We will close this issue when passed into master branch
